docker/r10k/docker-entrypoint.sh

```diff
@@ -1,11 +1,15 @@
 #! /bin/sh
 
-set -e
+set -ue
 
 for f in /docker-entrypoint.d/*.sh; do
   # Don't print out any messages here since this is a CLI container
-  chmod +x "$f"
-  "$f"
+  su "${PP_USER}" -- "${f}"
 done
 
-exec /usr/bin/r10k "$@"
+for f in /docker-custom-entrypoint.d/*.sh; do
+  # Don't print out any messages here since this is a CLI container
+  [ -x "${f}" ] && "${f}"
+done
+
+exec su "${PP_USER}" -c "/usr/bin/r10k $*"
```

Comment:
This is done because I have removed
```Dockerfile
USER puppet
```
from Dockerfile & release.Dockerfile

So to make sure that `r10k` is executed by the `${PP_USER}` (aka puppet) user I am using `su` because when `USER` in Dockerfile is not specified `root` is used. The reason for being the `root` user is, so that the scripts in `/docker-custom-entrypoint.d` is run as root because that is something I need.